impl/src/test/java/io/kroxylicious/testing/kafka/KafkaClusterTest.java

```diff
@@ -49,138 +63,134 @@ public class KafkaClusterTest {
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void kafkaClusterContainerKraftModeWithAuth(@DimensionMethodSource("versions") @DimensionMethodSource("clusterSizes") @KRaftCluster @SaslPlainAuth({
+            @SaslPlainAuth.UserPassword(user = "guest", password = "guest")
+    }) TestcontainersKafkaCluster cluster)
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void kafkaClusterContainerKraftModeSASL_SSL(@DimensionMethodSource("versions") @DimensionMethodSource("clusterSizes") @KRaftCluster @Tls @SaslPlainAuth({
+            @SaslPlainAuth.UserPassword(user = "guest", password = "guest")
+    }) TestcontainersKafkaCluster cluster)
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void kafkaClusterContainerKraftModeSSL(@DimensionMethodSource("versions") @DimensionMethodSource("clusterSizes") @KRaftCluster @Tls TestcontainersKafkaCluster cluster)
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void kafkaClusterContainerZookeeperModeWithAuth(@DimensionMethodSource("versions") @DimensionMethodSource("clusterSizes") @ZooKeeperCluster @SaslPlainAuth({
+            @SaslPlainAuth.UserPassword(user = "guest", password = "guest")
+    }) TestcontainersKafkaCluster cluster)
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void kafkaClusterContainerZookeeperModeSASL_SSL(@DimensionMethodSource("versions") @DimensionMethodSource("clusterSizes") @ZooKeeperCluster @Tls @SaslPlainAuth({
+            @SaslPlainAuth.UserPassword(user = "guest", password = "guest")
+    }) TestcontainersKafkaCluster cluster)
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
+    }
+
+    @TestTemplate
+    @ExtendWith(KafkaClusterExtension.class)
+    public void kafkaClusterContainerZookeeperModeSSL(@DimensionMethodSource("versions") @DimensionMethodSource("clusterSizes") @ZooKeeperCluster @Tls TestcontainersKafkaCluster cluster)
+            throws Exception {
+        verifyRecordRoundTrip(((BrokerCluster) brokerClusters[clusterSizesIndex++]).numBrokers(), cluster);
```

Comment:
Not an issue directly from this PR but we should change https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/15/files#diff-98fb5d27cd329bdf14c1cf95c9163c9d859206844b4ba1d59d04b6a8d73672ddR207 so that it expects `message` rather than the hard-coded string.